resolver/addressable_resolver.go

```diff
@@ -45,14 +50,16 @@ type RefResolverFunc func(ctx context.Context, ref *corev1.ObjectReference) (boo
 type URIResolver struct {
 	tracker       tracker.Interface
 	listerFactory func(schema.GroupVersionResource) (cache.GenericLister, error)
+	serviceLister corev1listers.ServiceLister
```

Comment:
If I understand this part right, listers from this factory returns `duck.Addressable` type:
https://github.com/knative/pkg/blob/c1f8225ddec49cbc3c81b9e4c14db360961f16e6/resolver/addressable_resolver.go#L65-L70

It would be great if we could reuse this factory, but I couldn't find a way to do that.